xla/service/gpu/matmul_utils.cc

```diff
@@ -620,62 +621,62 @@ Status RunGemm(const GemmConfig& config, se::DeviceMemoryBase lhs_buffer,
-                                 profile_result);
-    case BF16:
-      return DoGemm<Eigen::bfloat16>(
-          batch_size, m, n, k, lhs, rhs, output,
-          static_cast<Eigen::bfloat16>(config.alpha.real()),
-          static_cast<Eigen::bfloat16>(config.beta), stream, algorithm,
-          config.compute_precision, profile_result);
-    case F32:
-      return DoGemm<float>(batch_size, m, n, k, lhs, rhs, output,
-                           config.alpha.real(), config.beta, stream, algorithm,
-                           config.compute_precision, profile_result);
-    case F64:
-      return DoGemm<double>(batch_size, m, n, k, lhs, rhs, output,
-                            config.alpha.real(), config.beta, stream, algorithm,
-                            config.compute_precision, profile_result);
-    case C64:
-      return DoGemm<complex64>(batch_size, m, n, k, lhs, rhs, output,
-                               static_cast<complex64>(config.alpha),
-                               static_cast<complex64>(config.beta), stream,
-                               algorithm, config.compute_precision,
-                               profile_result);
-    case C128:
-      return DoGemm<complex128>(
-          batch_size, m, n, k, lhs, rhs, output, config.alpha,
-          static_cast<complex128>(config.beta), stream, algorithm,
-          config.compute_precision, profile_result);
-    default:
-      return InternalError(
-          "Unexpected GEMM dtype: %s",
-          primitive_util::LowercasePrimitiveTypeName(output_layout.dtype));
+#define TYPED_GEMM_COMPLEX(SCALENTYPE, ATYPE, BTYPE, CTYPE)                 \
+  if (operand_types == std::make_tuple(ATYPE, BTYPE, CTYPE)) {              \
+    using NativeScaleType =                                                 \
+        primitive_util::PrimitiveTypeToNative<SCALENTYPE>::type;            \
+    using NativeAType = primitive_util::PrimitiveTypeToNative<ATYPE>::type; \
+    using NativeCType = primitive_util::PrimitiveTypeToNative<CTYPE>::type; \
+    return DoGemm<NativeScaleType, NativeAType, NativeCType>(               \
+        batch_size, m, n, k, lhs, rhs, output,                              \
+        static_cast<NativeScaleType>(config.alpha),                         \
+        static_cast<NativeScaleType>(config.beta), stream, algorithm,       \
+        config.compute_precision, profile_result);                          \
+  }
+
+  if (output_layout.dtype == S32) {
+    if (!algorithm) algorithm = se::blas::kDefaultGemmAlgo;
+    return DoGemmWithAlgorithm<int32_t, int8_t, int32_t>(
+        batch_size, m, n, k, lhs, rhs, output,
+        static_cast<int32_t>(config.alpha.real()),
+        static_cast<int32_t>(config.beta), stream, *algorithm,
+        se::blas::kDefaultComputePrecision, profile_result);
   }
+
+  // TYPED_GEMM(F16, F16, F16, F16)
+  TYPED_GEMM(F32, BF16, BF16, BF16)
+  TYPED_GEMM(F32, F16, F16, F16)
+  // TYPED_GEMM(F32, S8, S8, F32)
```

Comment:
Why are some things commented out?

Reply:
removed. Cases are not supported.

Reply:
It seems you removed it here, but not in the TypesAreSupportedByLegacyCublas function. In some internal tests, we see this error being triggered:
"Unexpected GEMM dtype: s8 s8 f32".
Can you please double-check all supported cases again, whether they have a corresponding TYPED_GEMM or TYPED_GEMM_COMPLEX macro call?

Reply:
it is added back.